### PR TITLE
[semver:minor] Add support for toggling tests with ginkgo labels

### PR DESCRIPTION
**Pull request by nick-armstrong-helix**
## Background
The current `go-mod-ginkgo-v2` test orb can run two types of ginkgo tests, unit and acceptance. These test types are great and cover a wide range of use cases but it can cause confusion when trying to write functional and/or integration tests. This is because it is not very clear whether a functional and/or integration test should be binned into the unit or acceptance test category.

Furthermore extending the `go-mod-ginkgo-v2` job to support new test types is difficult because the job identifies acceptance tests by the directory name `acceptance/` and unit tests as everything else. In order for this job to add new tests a new test directory would need to be added for each test type, eg. `integration/`, `functional/`, etc, and this isn't very practical when the scope of the test is not project wide.

## What
This PR adds a ginkgo-v2-labels job that leverages the ginkgo label filtering system to select tests based on label filters. This solves the problem described in background by decoupling the test's location from how the test is run and instead tightly couples how tests are run to a label. This lets integration and functional tests to be located near the code they test and makes it easier to add new test types without having to change the project's directory structure.

## Summary
- Adds a ginkgo-v2-labels job and command
- Adds scripts to
  - determine the go env
  - install ginkgo v2
  - execute ginkgo with label filters

### CODE REVIEW

**Review by zubinmadon — COMMENTED** [review]
Nit: perhaps replace "flags" with "labels" or "label-filters" in the filenames, descpriptions, comments etc, to make it more clear how it differs from existing ginkgo jobs.

**Comment by nick-armstrong-helix** [comment]
⚠️ auto-merge enabled ⚠️

File: src/jobs/go-mod-ginkgo-v2-labels.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    type: string
+    default: "myhelix"
+  project_name:
+    description: "Project name, must match github repo name"
+    type: string
+  test_type:
+    description: The test type to run. Use this or label_filter_override, not both.
+    type: enum
+    enum: ["unit", "integration", "functional", "acceptance", "e2e", "performance", ""]
+    default: ""
```
Comment (JackyLiHelix) [inline src/jobs/go-mod-ginkgo-v2-labels.yml:36]: Same applies here.

**Review by JackyLiHelix — CHANGES_REQUESTED** [review]
Looks good overall. One minor improvement.

File: src/commands/ginkgo-v2-labels.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    type: string
+    default: "myhelix"
+  project_name:
+    description: "Project name, must match github repo name"
+    type: string
+  test_type:
+    description: The test type to run. Use this or label_filter_override, not both.
+    type: enum
+    enum: ["unit", "integration", "functional", "acceptance", "e2e", "performance", ""]
+    default: ""
```
[the remark this answers is not in the document]
Comment (nick-armstrong-helix) [reply src/commands/ginkgo-v2-labels.yml:36]: Nice catch! Thank you!

**Review by JackyLiHelix — APPROVED** [review]

**Pull request merged**